### DISCUSSION TITLE
Could you prevent users from installing plugins and only allow them to choose from ones that have API key specified in the .env file?

### DISCUSSION THREAD

**Discussion by igodsewer** [danny-avila/LibreChat#2132] [Q&A]
Unfortunately, I couldn't find any relevant info, so here I am with the question. Thanks in advance!

**Comment by igodsewer** [comment]
To be more precise, I'm looking to disable all plugins except DALLE3

**Reply by fuegovic** [reply]
You can remove some of them from `\api\app\clients\tools\manifest.json`
The other ones can be deleted from `\api\app\clients\tools\.well-known\`

**Reply by igodsewer** [reply]
Well, that's a dirty way, I guess. Is it possible via .env file?

Thanks,

**Reply by danny-avila** [reply]
not yet, this has been requested a lot so getting to it this week.

**Reply by fuegovic** [reply]
haha indeed, it's not the most elegant solution 😅

Should you decide to use the suggested workaround, ensure you back up the files as a precaution.

**Reply by philipbrembeck** [reply]
@danny-avila Are there any updates on this?

**Reply by danny-avila** [reply]
https://docs.librechat.ai/install/configuration/custom_config.html#filteredtools

**Reply by philipbrembeck** [reply]
@danny-avila I guess I'm a bit lost here, but using this in the current v0.7.1, with the config version 1.0.6,  I run into:
```bash
2024-05-03 14:05:14 error: Invalid custom config file at /app/librechat.yaml [
  {
    "code": "unrecognized_keys",
    "keys": [
      "filteredTools"
    ],
    "path": [],
    "message": "Unrecognized key(s) in object: 'filteredTools'"
  }
]
```

**Reply by danny-avila** [reply]
@philipbrembeck this means you haven't updated correctly. What did you do to update, are you running docker?

**Reply by danny-avila** [reply]
https://docs.librechat.ai/install/installation/docker_compose_install.html#updating-librechat

**Reply by philipbrembeck** [reply]
I'm running a Helm chart I'm currently creating, so that may be the issue. Thanks for the hint!

**Reply by prononext** [reply]
would be really nice to see that functionality soon

**Reply by danny-avila** [reply]
@prononext already implemented: https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/config#filteredtools
